### ISSUE TITLE
Access Denied #49 after query

### ISSUE THREAD

**Issue by zaraki12345** [pardot/api-docs#188]
Hi there!

From few days got problem with API. Using version 3. Using API is update and method that i'm using is supported. 
That's how look the url query:
https://pi.pardot.com/api/form/version/3/do/query?user_key=XXXX&api_key=XXXXX

I'm receiving the api key but getting message error:
<rsp stat="fail" version="1.0">
<err code="49">Access Denied</err>
</rsp>
Credentials are passed via supported method, user got all permission. It was working but suddenly stopped.
here the request
`
private function httpRequest($url, $postData = [])
  {
    $ch = curl_init();

    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_HEADER, false); 
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); 
    curl_setopt($ch, CURLOPT_POST, true); 
    curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);

    $response = curl_exec($ch);

    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

    curl_close($ch);

    if($httpCode == 200) {
      return simplexml_load_string($response);
    }
  }
}
`

**Comment by adelawalla** [comment]
Passing the user_key and api_key in the query string is no longer allowed: http://developer.pardot.com/. If you move the credentials to the [Authorization header](http://developer.pardot.com/#using-the-api), it should work again.

**Comment by zaraki12345** [comment]
Please could you provide some samples in php because your sample is using
query strings...

wt., 16 lip 2019 o 15:51 Adil Delawalla <notifications@github.com>
napisał(a):

**Comment by jpearson-sfdc** [comment]
@zaraki12345 Thanks for bringing up the issue with the docs; we'll need to update the code example in the docs. Below are the instructions to unblock you now.

Given the function you specified above, you need to set the `CURLOPT_HTTPHEADER` option like the following. Make sure to replace the `XXXX` below with your user_key and api_key.

```php
private function httpRequest($url, $postData = [])
{
    $ch = curl_init();

    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_HEADER, false); 
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); 
    curl_setopt($ch, CURLOPT_POST, true); 
    curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'Authorization: Pardot user_key=XXXX,api_key=XXXX'
    ]);

    $response = curl_exec($ch);

    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

    curl_close($ch);

    if($httpCode == 200) {
        return simplexml_load_string($response);
    }
}
```

**Comment by zaraki12345** [comment]
`  private function httpRequest($url, $postData = [])
  { 
      $userKey = $this->userKey;
      $userApiKey = $this->userApiKey;
      $ch = curl_init();
      curl_setopt($ch, CURLOPT_URL, $url);
      curl_setopt($ch, CURLOPT_HEADER, false); 
      curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); 
      curl_setopt($ch, CURLOPT_POST, true); 
      curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);
      curl_setopt($ch, CURLOPT_HTTPHEADER, [
          'Authorization: 
          Pardot user_key=' . $userKey . ',
          api_key=' . $userApiKey  
      ]);

      $response = curl_exec($ch);

      $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

      curl_close($ch);

      if($httpCode == 200) {
          return simplexml_load_string($response);
      }
  }`

For me this solution work's. I lost 1,5 day for looking error in my code :/ Please update ASAP documentation for future people. Because a lot of forms for my client didnt working and he lost some leads :/

**Comment by rappadv** [comment]
Can I get some help in C#? There is no documentation on this. I need to get the API key first.

We were using this:
var jsonKey = wc.DownloadString("https://pi.pardot.com/api/login/version/4?format=json&email=" + pardotLoginEmail + "&password=" + pardotLoginPassword + "&user_key=" + pardotLoginKey);

I tried adding this:
wc.Headers.Add("Authorization", "Pardot email=" + pardotLoginEmail + ", password=" + pardotLoginPassword + ", user_key=" + pardotLoginKey);

**Comment by rappadv** [comment]
Never mind. I guess it is still ok to pass credentials to get the API key.

**Comment by jpearson-sfdc** [comment]
This was fixed with commit 035df12ad7d76d43daf5e6884f04f9dad9e8b242
